[tn48m/tn48m-dn series] Upgrade kernel version to 5.10

chenglin-tsai opened the pull request:
Upgrade kernel version to 5.10 on tn48m and tn48m-dn series.

Test logs:
[dentOS_tn4810m_kernel_5.10_onlpdump_2021_0218.log](https://github.com/dentproject/dentOS/files/6000656/dentOS_tn4810m_kernel_5.10_onlpdump_2021_0218.log)
[dentOS_tn48m_kernel_5.10_onlpdump_2021_0218.log](https://github.com/dentproject/dentOS/files/6000657/dentOS_tn48m_kernel_5.10_onlpdump_2021_0218.log)
[dentOS_tn48m-poe_kernel_5.10_onlpdump_2021_0218.log](https://github.com/dentproject/dentOS/files/6000658/dentOS_tn48m-poe_kernel_5.10_onlpdump_2021_0218.log)


Signed-off-by: Chenglin Tsai <chenglin.tsai@deltaww.com>

paulmenzel commented on packages/platforms/delta/arm64/tn48m-dn/tn4810m-dn/platform-config/r0/src/python/arm64_delta_tn4810m_dn_r0/__init__.py:
```diff
@@ -37,8 +37,8 @@ def baseconfig(self):
-        self.insmod("prestera_sw.ko")
-        self.insmod("prestera_pci.ko")
+        # Insert Marvell prestera modules by only probing prestera_pci module
+        # Suggested by Taras Chornyi
```
Please put that into the commit message.

chenglin-tsai replied:
Done and thanks.

paulmenzel commented on packages/platforms/delta/arm64/tn48m-dn/tn4810m-dn/platform-config/r0/src/python/arm64_delta_tn4810m_dn_r0/__init__.py:
```diff
@@ -37,8 +37,8 @@ def baseconfig(self):
-        self.insmod("prestera_pci.ko")
+        # Insert Marvell prestera modules by only probing prestera_pci module
+        # Suggested by Taras Chornyi
+        subprocess.call('modprobe prestera_pci', shell=True)
```
Why is `self.insmod("prestero_pci.ko")` not working anymore?

chenglin-tsai replied:
> Why is `self.insmod("prestero_pci.ko")` not working anymore?

Hi Paul,

Because the name of prestera driver is changed from "prestera_sw.ko" to "prestera.ko", it will be better if we use modprobe to probe prestero_pci driver only, modprobe will automatically help us to insert the necessary drivers, like prestera.ko here.

For more details, you can refer to the discussion: https://github.com/dentproject/dentOS/pull/43#issuecomment-777704551

paulmenzel replied:
I wonder if a method `modprobe` should be added to `packages/base/all/vendor-config-onl/src/`, so only `self.modprobe("prestero")` needs to be used.

chenglin-tsai replied:
> I wonder if a method `modprobe` should be added to `packages/base/all/vendor-config-onl/src/`, so only `self.modprobe("prestero")` needs to be used.

Yes, the self.modprobe function can be implemented in file: "packages\base\all\vendor-config-onl\src\python\onl\platform\base.py" in the future if needed.

paulmenzel commented on packages/base/any/kernels/5.10-lts/patches/series.arm64:
```diff
@@ -16,3 +16,5 @@
 0021-tn48m-add-sfp-eeprom-support-to-ethtool.patch
 #0022-marvell-wa-tx-disable.patch
+0023-delta-tn48m-dn-series-dts.patch
+
```
Please remove the blank line at the end of the file.

chenglin-tsai replied:
Done and thanks.

paulmenzel commented on packages/platforms/delta/arm64/tn48m-dn/tn48m-dn/platform-config/r0/src/python/arm64_delta_tn48m_dn_r0/__init__.py:
```diff
@@ -26,9 +26,9 @@ def baseconfig(self):
-        # Insert prestera kernel module
-        self.insmod("prestera_sw.ko")
-        self.insmod("prestera_pci.ko")
+        # Insert Marvell prestera modules by only probing prestera_pci module
+        # Suggested by Taras Chornyi
+        subprocess.call('modprobe prestera_pci', shell=True)
```
Ditto as above.

paulmenzel commented on packages/platforms/delta/arm64/tn48m-dn/tn48m-poe-dn/platform-config/r0/src/python/arm64_delta_tn48m_poe_dn_r0/__init__.py:
```diff
@@ -35,8 +35,8 @@ def baseconfig(self):
-        # Insert prestera kernel module
-        self.insmod("prestera_sw.ko")
-        self.insmod("prestera_pci.ko")
+        # Insert Marvell prestera modules by only probing prestera_pci module
+        # Suggested by Taras Chornyi
+        subprocess.call('modprobe prestera_pci', shell=True)
```
Ditto as above.

paulmenzel commented on packages/platforms/delta/arm64/tn48m/tn4810m/platform-config/r0/src/python/arm64_delta_tn4810m_r0/__init__.py:
```diff
@@ -41,8 +41,8 @@ def baseconfig(self):
-        # Insert prestera kernel module
-        self.insmod("prestera_sw.ko")
-        self.insmod("prestera_pci.ko")
+        # Insert Marvell prestera modules by only probing prestera_pci module
+        # Suggested by Taras Chornyi
+        subprocess.call('modprobe prestera_pci', shell=True)
```
Ditto as above. (I won’t comment on the other occurrences.)

paulmenzel reviewed (commented):
Is ` packages/base/any/kernels/5.10-lts/patches/0023-delta-tn48m-dn-series-dts.patch` just a copy from the 5.6 version, or did you have to update it?

chenglin-tsai commented:
> Is ` packages/base/any/kernels/5.10-lts/patches/0023-delta-tn48m-dn-series-dts.patch` just a copy from the 5.6 version, or did you have to update it?

The 'packages/base/any/kernels/5.10-lts/patches/0023-delta-tn48m-dn-series-dts.patch' is just a copy from kernel 5.6.

paulmenzel commented:
Here are the out-of-Linux-kernel-tree switch driver messages.

```
[   83.823249] Loading Marvell Prestera Switch Driver
[   83.846307] AC3x B2B 98DX3265 0000:01:00.0: FW version '2.8.0'
[   83.852192] AC3x B2B 98DX3265 0000:01:00.0: Driver version '2.8.0'
[   83.858408] AC3x B2B 98DX3265 0000:01:00.0: Loading prestera FW image ...
[   84.070193] sfp sfp-52: module AVAGO            ABCU-5712RZ-HP1  rev      sn 2Y1139MD70       dc 111006
[   86.820978] AC3x B2B 98DX3265 0000:01:00.0: Prestera Switch FW is ready
[   97.137909] AC3x B2B 98DX3265 0000:01:00.0: Initialized Switch device
[   99.034257] prestera_pci: Registered Marvell Prestera PCI driver
```

paulmenzel commented:
> > Is ` packages/base/any/kernels/5.10-lts/patches/0023-delta-tn48m-dn-series-dts.patch` just a copy from the 5.6 version, or did you have to update it?
> 
> The 'packages/base/any/kernels/5.10-lts/patches/0023-delta-tn48m-dn-series-dts.patch' is just a copy from kernel 5.6.

Could you please add that to the git commit message?

chenglin-tsai commented:
> > > Is ` packages/base/any/kernels/5.10-lts/patches/0023-delta-tn48m-dn-series-dts.patch` just a copy from the 5.6 version, or did you have to update it?
> > 
> > 
> > The 'packages/base/any/kernels/5.10-lts/patches/0023-delta-tn48m-dn-series-dts.patch' is just a copy from kernel 5.6.
> 
> Could you please add that to the git commit message?

Sure. Added.

chenglin-tsai commented:
Since the 'self.modprobe' method is merged into codebase [1], so update the '__init__.py' in tn48m/tn48m-dn platforms to use 'self.modprobe' method to probe Marvell prestera drivers. And also update the commit message.

[1]: https://github.com/dentproject/dentOS/pull/51

Mickey201 commented on packages/base/any/kernels/5.10-lts/patches/series.arm64:
```diff
@@ -16,3 +16,4 @@
 0020-accton-as5114.patch
 0021-tn48m-add-sfp-eeprom-support-to-ethtool.patch
 #0022-marvell-wa-tx-disable.patch
```
This Line:
#0022-marvell-wa-tx-disable.patch
should be removed and you can use:   
**0022**-delta-tn48m-dn-series-dts.patch

chenglin-tsai replied:
@Mickey201, No problem, updated in https://github.com/dentproject/dentOS/pull/45/commits/e63345162b252ee5bc6000a2fd4a7e9e5cce2453. Thanks for the suggestions.

Mickey201 reviewed (commented):
Chenglin, 
Did you tested it with Marvell Switchdev 5.10 support?
I did a review - it looks ok.

chenglin-tsai commented:
> Chenglin,
> Did you tested it with Marvell Switchdev 5.10 support?
> I did a review - it looks ok.

Yes, in the test log, I saw the Marvell firmware 2.8.0 is successfully loaded in kernel 5.10.

[dentOS_tn4810m_kernel_5.10_onlpdump_2021_0218.log](https://github.com/dentproject/dentOS/files/6000656/dentOS_tn4810m_kernel_5.10_onlpdump_2021_0218.log)

```
[   85.087437] Aldrin2 0000:01:00.0: FW version '2.8.0'
[   85.095775] Aldrin2 0000:01:00.0: Driver version '2.8.0'
[   85.095780] Aldrin2 0000:01:00.0: Loading prestera FW image ...
[…]
[   88.040225] Aldrin2 0000:01:00.0: Prestera Switch FW is ready
[  103.545663] Aldrin2 0000:01:00.0: Initialized Switch device
[  104.981497] prestera_pci: Registered Marvell Prestera PCI driver
[…]
[  121.036016] Aldrin2 0000:01:00.0 swp12: renamed from eth11
[  121.097156] Aldrin2 0000:01:00.0 swp19: renamed from eth18
[  121.133812] Aldrin2 0000:01:00.0 swp18: renamed from eth17
[  121.166944] Aldrin2 0000:01:00.0 swp16: renamed from eth15
[  121.235866] Aldrin2 0000:01:00.0 swp2: renamed from eth1
[  121.303082] Aldrin2 0000:01:00.0 swp20: renamed from eth19
[  121.342853] Aldrin2 0000:01:00.0 swp23: renamed from eth22
[  121.376274] Aldrin2 0000:01:00.0 swp15: renamed from eth14
[  121.427043] Aldrin2 0000:01:00.0 swp1: renamed from eth0
[  121.466958] Aldrin2 0000:01:00.0 swp13: renamed from eth12
[  121.499850] Aldrin2 0000:01:00.0 swp11: renamed from eth10
[  121.544351] Aldrin2 0000:01:00.0 swp3: renamed from eth2
[  121.570874] Aldrin2 0000:01:00.0 swp22: renamed from eth21
[  121.603356] Aldrin2 0000:01:00.0 swp21: renamed from eth20
[  121.635145] Aldrin2 0000:01:00.0 swp27: renamed from eth26
[  121.667433] Aldrin2 0000:01:00.0 swp14: renamed from eth13
[  121.700835] Aldrin2 0000:01:00.0 swp26: renamed from eth25
[  121.743227] Aldrin2 0000:01:00.0 swp4: renamed from eth3
[  121.775066] Aldrin2 0000:01:00.0 swp33: renamed from eth32
[  121.807846] Aldrin2 0000:01:00.0 swp31: renamed from eth30
[  121.846981] Aldrin2 0000:01:00.0 swp30: renamed from eth29
[  121.880769] Aldrin2 0000:01:00.0 swp32: renamed from eth31
[  121.942627] Aldrin2 0000:01:00.0 swp29: renamed from eth28
[  121.967012] Aldrin2 0000:01:00.0 swp24: renamed from eth23
[  122.015509] Aldrin2 0000:01:00.0 swp39: renamed from eth38
[  122.039552] Aldrin2 0000:01:00.0 swp5: renamed from eth4
[  122.067894] Aldrin2 0000:01:00.0 swp34: renamed from eth33
[  122.098622] Aldrin2 0000:01:00.0 swp38: renamed from eth37
[  122.126614] Aldrin2 0000:01:00.0 swp25: renamed from eth24
[  122.150856] Aldrin2 0000:01:00.0 swp35: renamed from eth34
[  122.182784] Aldrin2 0000:01:00.0 swp28: renamed from eth27
[  122.214640] Aldrin2 0000:01:00.0 swp40: renamed from eth39
[  122.243160] Aldrin2 0000:01:00.0 swp42: renamed from eth41
[  122.267431] Aldrin2 0000:01:00.0 swp17: renamed from eth16
[  122.290987] Aldrin2 0000:01:00.0 swp43: renamed from eth42
[  122.323213] Aldrin2 0000:01:00.0 swp44: renamed from eth43
[  122.351256] Aldrin2 0000:01:00.0 swp37: renamed from eth36
[  122.378654] Aldrin2 0000:01:00.0 swp36: renamed from eth35
[  122.399171] Aldrin2 0000:01:00.0 swp45: renamed from eth44
[  122.431262] Aldrin2 0000:01:00.0 swp47: renamed from eth46
[  122.468878] Aldrin2 0000:01:00.0 swp48: renamed from eth47
[  122.508975] Aldrin2 0000:01:00.0 swp6: renamed from eth5
[  122.546450] Aldrin2 0000:01:00.0 swp46: renamed from eth45
[  122.571406] Aldrin2 0000:01:00.0 swp41: renamed from eth40
[  122.595387] Aldrin2 0000:01:00.0 swp8: renamed from eth7
[  122.634863] Aldrin2 0000:01:00.0 swp7: renamed from eth6
[  122.660285] Aldrin2 0000:01:00.0 swp9: renamed from eth8
[  122.696325] Aldrin2 0000:01:00.0 swp10: renamed from eth9
```

taraschornyiplv commented:
@sonoble As pointed by @paulmenzel  in #44 commits from the pull request will be squashed and merged to main branch as one commit. Is it possible to enable [Rebase and Merge](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-request-merges#rebase-and-merge-your-pull-request-commits) option
